src/app/cuestionario/cuestionario.component.ts

```diff
@@ -9,24 +9,73 @@ import { DatosService } from '../datos.service';
   styleUrl: './cuestionario.component.css'
 })
 export class CuestionarioComponent {
-  preguntaNombre: Array<string> = [];
+  numeroDePreguntas: number = 4;
+  numeroDeRespuestas: number = 4;
+
+  preguntas: Record<string, Array<string>> = {
+    "preguntaNombre": [],
+    "preguntaPeso": [],
+    "preguntaNombreDos": [],
+    "preguntaAltura": [],
+  }
+
+  respuestasCorrectas: Array<number> = [];
+  respuestasUsuario: Array<boolean> = [];
 
-  respuestaCorrecta: Array<number> = [];
 
 
   constructor(
     public DatosService: DatosService,
   ) {
-    for (var i = 0; i <= this.DatosService.numerosPokemonsVisibles; ++i) {
-      this.preguntaNombre[i] = this.DatosService.listaPokemonsCuestionario[i].nombre;
+    for (var i = 0; i <= this.numeroDeRespuestas - 1; ++i) {
+      this.preguntas["preguntaNombre"][i] = this.DatosService.listaPokemonsCuestionario[i].nombre;
+    }
+    for (var i = 0; i <= this.numeroDeRespuestas - 1; ++i) {
+      this.preguntas["preguntaPeso"][i] = this.DatosService.listaPokemonsCuestionario[i].peso.toString();
+    }
+    for (var i = 0; i <= this.numeroDeRespuestas - 1; ++i) {
+      this.preguntas["preguntaNombre"][i] = this.DatosService.listaPokemonsCuestionario[i].nombre;
+    }
+    for (var i = 0; i <= this.numeroDeRespuestas - 1; ++i) {
+      this.preguntas["preguntaAltura"][i] = this.DatosService.listaPokemonsCuestionario[i].altura.toString();
+    }
```

Comment:
No veo por qué no podemos hacerlo todo en un único `for`

```suggestion
    for (var i = 0; i < this.numeroDeRespuestas; ++i) {
      this.preguntas["preguntaNombre"][i] = this.DatosService.listaPokemonsCuestionario[i].nombre;
      this.preguntas["preguntaPeso"][i] = this.DatosService.listaPokemonsCuestionario[i].peso.toString();
      this.preguntas["preguntaNombre"][i] = this.DatosService.listaPokemonsCuestionario[i].nombre;
      this.preguntas["preguntaAltura"][i] = this.DatosService.listaPokemonsCuestionario[i].altura.toString();
    }
```
Además creo que el segundo `preguntaNombre` es `preguntaNombreDos`